test/core/test_object_3d.rb

```diff
@@ -151,6 +151,40 @@ def test_to_json_with_children
     }, object.to_json)
   end
 
+  def test_to_json_with_mesh_child
+    object = Mittsu::Object3D.new
+    geometry = Mittsu::BoxGeometry.new(1.0, 1.0, 1.0)
+    material = Mittsu::MeshBasicMaterial.new(color: 0x00ff00)
+    object.add(child = Mittsu::Mesh.new(geometry, material))
+    # Check object basics
+    json = object.to_json
+    assert_equal('Mesh', json['object']['children'][0]['type'])
+    # Check geometry node and uuid references
+    assert_equal('Geometry', json['geometries']['children'][0]['type'])
```

Comment:
@danini-the-panini To make sure this is right, I'm adding specific tests for mesh and line children, and I'm not sure if the behaviour I'm going for is correct. 

Do ALL geometries and materials live in a single array at the root of the JSON, no matter the object nesting? Then the objects themselves reference them by UUIDs further down in the tree? So, the geometry and material for an object are NOT contained anywhere under the `object/children` array?

That's what it looks like it should be, from things like the Line and Mesh `test_to_json` tests, but I thought I'd check.

Reply:
I think all the objects exist in the `children` array where they're nested

Reply:
oh i see, i misunderstood.
yeah there should just be the top level materials and geometries, then the children objects just reference them by uuid.

Reply:
Great, thanks! It was a difficult question to phrase, no worries :)